pkg/sql/sqlstats/ssmemstorage/ss_mem_storage.go

```diff
@@ -446,7 +441,7 @@ func (s *stmtStats) mergeStatsLocked(statistics *appstatspb.CollectedStatementSt
 	s.mu.data.Add(&statistics.Stats)
 
 	// Setting all metadata fields.
-	if s.mu.data.SensitiveInfo.LastErr == "" && statistics.Key.Failed {
+	if s.mu.data.SensitiveInfo.LastErr == "" && statistics.Stats.SensitiveInfo.LastErr != "" {
```

Comment:
@xinhaoz What do you think about this updated conditional? If the LastErr is empty, and the stats we're merging has its LastErr populated, we take the LastErr from the stats being merged.